modules/ROOT/pages/api-gateway-encryption-mule4.adoc

```diff
@@ -0,0 +1,252 @@
+
+If you run into issues when passing your encrypted credentials to Anypoint Platform, it's recommended to check that the provided credentials are valid. You can use the Secure Properties Tool described earlier to decrypt the values back to the originals and compare them with the ones displayed in the platform:
+
+. Run the Secure Properties Tool to decrypt your credentials:
++
+[source]
+--
+> java -jar secure-properties-tool.jar string decrypt AES CBC <yourEncryptionKey> <>
+--
++
+. Compare the decrypted value with the one in Anypoint Platform. +
+See xref:org-credentials-config-mule4.adoc#obtaining-credentials[Obtaining Credentials] for information on how to verify your platform's credentials.
+
+== Supported Properties
+
+The following `wrapper.conf` properties support encryption:
+
+[%header%autowidth.spread,cols="a,a,a"]
+|===
+Property | | Description
+| Client ID | `anypoint.platform.client_id` | Your Platform Client ID
+| Client Secret | `anypoint.platform.client_secet` | Your Platform Client Secret
+| Proxy Password | `anypoint.platform.proxy_password` | Password of the proxy configuration to connect to Platform
+|===
+
+== Policies Encryption
+
+The configuration for your API policies is stored in the `policies` directory, and some policies may contain sensitive information that you want to hide from unauthorized parties, such as the password field for the `Basic Authentication` header. +
+Sensitive values in your provided API policies are marked as such in the API Manager UI for policy configuration. When creating a custom policy, you can define which values are sensitive.
+
+== Configuring Policy Encryption
+
+To enable encryption in policies, you need to provide two system properties in your `wrapper.conf` file:
+
+* `anypoint.platform.encryption_key`. +
+The encryption key for your policy.
++
+[CAUTION]
+--
+If you already have all your policies encrypted and then needed to restart the runtime without an encryption key, all policies will be downloaded again, and all the sensitive information won't be encrypted.
+--
+* `anypoint.platform.encryption_mode`. +
+This property can take two values:
++
+[%autowidth.spread,cols="a,a"]
+|===
+| `FULL` | All values are encrypted.
+| `SENSITIVE_ONLY` | Only values that you marked as `sensitive` are encrypted.
+|===
++
+You can change the encryption mode after applying one. The files in your `policies` directory will regenerate according to your new encryption mode.
++
+[CAUTION]
+--
+If you are changing from `FULL` to `SENSITIVE_ONLY`, any parameter not marked as `sensitive` will be shown as plain text in your filesystem.
+--
+
+== Policies That Support Encryption
+
+Below is a list of all policies that support encryption in their values, along with their minimum required version to support this feature. +
+Earlier versions of these policies, and policies not listed below will not encrypt the policies values even when the gateway capability detects and encryption key in its configuration.
+
```

Comment:
Typo, it should be "even when the gateway capability detects **an** encryption key in its configuration."

Reply:
Thanks!  Fixed in 5bb7bff